contrib/scalar/scalar.c

```diff
@@ -21,24 +21,28 @@ static int is_unattended(void) {
 
 /**
  * Remove the deepest subdirectory in the provided path string. Path must not
- * include a trailing path separator.
+ * include a trailing path separator. Returns 1 if parent directory found,
+ * otherwise 0.
  */
-static void strbuf_parentdir(struct strbuf *buf)
+static int strbuf_parentdir(struct strbuf *buf)
 {
+	size_t len = buf->len;
 	size_t offset = offset_1st_component(buf->buf);
 	char *path_sep = find_last_dir_sep(buf->buf + offset);
 	strbuf_setlen(buf, path_sep ? path_sep - buf->buf : offset);
+
+	return (buf->len < len);
 }
 
 /**
  * Given an absolute path at or below the enlistment root, find the
- * enlistment root and working directory
+ * enlistment root and working directory. Returns 1 if enlistment found,
+ * otherwise 0.
```

Comment:
The function name `find_enlistment()` sounds to me as if we expect to find an enlistment, i.e. it is not a "yes/no?" question. Therefore, I would let the function return `0` in the regular case, and `-1` if no enlistment was found.

Indeed, I would potentially even go further and show an error message if no enlistment was found (this would require a copy of `path` to be stored, and released, though).

Reply:
Compared to `strbuf_parentdir`, I'm more convinced that this should be represented with "success/error" than "true/false" (since the input `path` is mangled in the process). However, `find_enlistment(...)` ended up only having one usage in this file - would it be better to just re-integrate it with `setup_enlistment_directory(...)`? That function is already meant to throw an error if the enlistment isn't found (as opposed to `find_enlistment(...)`, which I had intended _not_ throw an error in case it was used only to get information about the enlistment).

Reply:
> would it be better to just re-integrate it with `setup_enlistment_directory(...)`?

That's a good idea!